extensions/spark/kyuubi-spark-authz/src/main/scala/org/apache/kyuubi/plugin/spark/authz/PrivilegesBuilder.scala

```diff
@@ -65,49 +65,65 @@ object PrivilegesBuilder {
 
     def mergeProjection(table: Table, plan: LogicalPlan): Unit = {
       if (projectionList.isEmpty) {
-        privilegeObjects += PrivilegeObject(table, plan.output.map(_.name))
+        plan match {
+          case pvm: PermanentViewMarker if pvm.output.isEmpty =>
+            privilegeObjects += PrivilegeObject(table, pvm.catalogTable.schema.map(_.name))
+          case _ =>
+            privilegeObjects += PrivilegeObject(table, plan.output.map(_.name))
+        }
       } else {
         val cols = (projectionList ++ conditionList).flatMap(collectLeaves)
           .filter(plan.outputSet.contains).map(_.name).distinct
         privilegeObjects += PrivilegeObject(table, cols)
       }
     }
 
+    // Should execute this for all plan and skip PermanentViewMaker
```

Comment:
how about we add 
```scala
    plan.expressions.foreach {
      case subquery: SubqueryExpression =>
        buildQuery(subquery.plan, privilegeObjects, projectionList, conditionList, spark)
    }
```
here?

Reply:
Yea, just changed, since we also need to support dynamic partition pruning

Reply:
Maybe we should move it to the end of this method? The final project list should be calculated first.

Reply:
> Maybe we should move it to the end of this method? The final project list should be calculated first.

Done